x-pack/plugins/licensing/public/plugin.test.ts

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Observable } from 'rxjs';
+import { bufferCount, first, take, skip } from 'rxjs/operators';
+import { ILicense } from '../common/types';
+import { License } from '../common/license';
+import { licenseMerge } from '../common/license_merge';
+import { API_ROUTE, LICENSING_SESSION, LICENSING_SESSION_SIGNATURE } from '../common/constants';
+import { delay } from '../common/delay';
+import { Plugin } from './plugin';
+import { mockHttpInterception, setup, setupOnly } from './__fixtures__/setup';
+
+describe('licensing plugin', () => {
+  let plugin: Plugin;
+  let license: ILicense;
+
+  afterEach(async () => {
+    await plugin.stop();
+    sessionStorage.clear();
+  });
+
+  test('returns instance of licensing setup', async () => {
+    ({ plugin, license } = await setup());
+    expect(license).toBeInstanceOf(License);
+  });
+
+  test('intercepted HTTP request changes license signature', async () => {
+    const { coreSetup, plugin: _plugin } = await setupOnly();
+
+    plugin = _plugin;
+    mockHttpInterception(coreSetup);
+
+    const { license$ } = await plugin.setup(coreSetup);
+
+    await license$
+      .pipe(
+        skip(1),
+        take(1)
+      )
+      .toPromise();
+    await coreSetup.http.get('/api/fake');
+
+    expect(plugin.sign()).toBe('fake-signature');
+  });
+
+  test('intercepted HTTP request causes a refresh if the signature changes', async () => {
+    const { coreSetup, plugin: _plugin } = await setupOnly();
+
+    plugin = _plugin;
+    mockHttpInterception(coreSetup);
+
+    const licensingSetup = await plugin.setup(coreSetup);
+
+    await licensingSetup.license$
+      .pipe(
+        skip(1),
+        take(1)
+      )
+      .toPromise();
+
+    const refresh = jest.spyOn(licensingSetup, 'refresh');
+
+    await coreSetup.http.get('/api/fake');
+
+    expect(refresh).toHaveBeenCalled();
+  });
+
+  test('calling refresh triggers fetch', async () => {
+    const { coreSetup, plugin: _plugin, license$, refresh } = await setup();
+
+    // We create a dummy subscription to ensure that calls to refresh actually
+    // get triggered in the observable.
+    license$.subscribe(() => {});
+    plugin = _plugin;
+    coreSetup.http.get.mockClear();
+    refresh();
+    await delay(200);
```

Comment:
why 200ms.? 
also it may lead to flaky behaviour:  https://github.com/elastic/kibana/pull/47641

Reply:
This is different than the poller tests. The delay is merely a sanity courtesy to ensure we give enough time to the event loop to hit the `http.get` call.